ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py

```diff
@@ -2080,6 +2081,32 @@ def test_torch_instance_device(dtype_and_x, as_variable, native_array):
     )
 
 
+# is_cuda
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+    ),
+    copy=st.booleans(),
+    device=st.booleans(),
+)
+def test_torch_instance_is_cuda(dtype_and_x, copy, device):
```

Comment:
You should use the method new_ones(self, size, *, dtype=None, device=None, requires_grad=False) of Tensor to generate data to be tested and use the "given" section to generate the device to be assigned to the generated Tensor. In this way, you can make this test a more standard hypothesis test instead of assert test.

Reply:
Can I get an example of Tensor().new_ones()? I am facing a problem with assigning device to new_ones.

Reply:
The new_ones function is another instance function in the same torch/tensor file that you are working on, I think you can aslo search for it in the torch(the real one)'s document as there must be a real counter part there.

Reply:
By the way, as this is an instance method, you should check other tests in the same test file you are working on as examples as they need extra initialization-related arguments.

Reply:
As mentioned in the previous comment, I have tried with torch.device and ivy.Device. I keep getting error for both of these cases. The error specifically being: 

"function has no attribute replace"

This error is shown at line 28 of ivy/functional/torch/device.py

Reply:
Please report this error to the functions/device channel of the discord.

Reply:
done

Reply:
@Daniel4078 I have merged the updated code from master branch to is_cuda branch. I keep getting the following error:

![image](https://user-images.githubusercontent.com/34241842/202286123-bb0ac63a-f211-4b3e-b10c-4f64576380d3.png)

Could you give me some insight regarding this?

Reply:
Well, can you locate the arctan_ function?

Reply:
Also, due to the recent refactoring of ivy code base, most of the tests are going to fail on their own.

Reply:
Hello @everyone, we just merged a huge refactor for the test, here are few major updates, we now have 4 new decorators instead of handle_cmd_line_args :

1. handle_function: used for Ivy functional API
2. handle_method: used for Ivy methods
3. handle_frontend_function: used for frontend functions
4. handle_frontend_method: used for frontend methods, methods tests are now skipped temporarily, sorry about that, will be fixed ASAP!

Some tests may start failing, as now Hypothesis are able to search into more example, If you notice unexpected failures or weird bugs please feel free to post it on the Testing channel.

More details will be announced later, you can read more about the new decorators and how they're used here:
https://github.com/unifyai/ivy/blob/d666a0fbe0106570402ed86fd87858c3f5754bb3/docs/partial_source/deep_dive/ivy_tests.rst

Thanks all!